src/core/socket.c

```
@@ -1027,16 +1027,13 @@ static void socket_close_fds(Socket *s) {
 
 DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(Socket*, socket_close_fds, NULL);
 
-#define log_socket_option_errno(s, e, option)                                          \
-        ({                                                                             \
-                int _e_ = (e);                                                         \
-                log_unit_full_errno(                                                   \
-                                UNIT(s),                                               \
-                                ERRNO_IS_NOT_SUPPORTED(_e_) ? LOG_DEBUG : LOG_WARNING, \
-                                _e_,                                                   \
-                                "Failed to set %s socket option, ignoring: %m",        \
-                                option);                                               \
-        })
+#define log_socket_option_errno_full(s, e, debug, option)                               \
```

> I'm still not convinced that it's a good idea to suppress
the EOPNOTSUPP everywhere, as it could also mean that
the option is set for the wrong type of socket.

EOPNOTSUPP should not be caused by users fault, but caused by our code, or the kernel is simply old. So, I'd like to unconditionally suppress error message in such cases.

> And two ERRNO_IS_NEG_PRIVILEGE conditions are inadvertently
dropped.

Ouch.

---

> EOPNOTSUPP should not be caused by users fault, but caused by our code, or the kernel is simply old. So, I'd like to unconditionally suppress error message in such cases.

Hmm? Why? If user enables tcp options on AF_UNIX sockets they will get EOPNOTSUPP, as expected? We shouldn't hide those errors?

---

Re-reading the original issue #39792, the warning comes when both AF_UNIX and AF_INET6 sockets are configured in a single .socket file.

So, such configuration is allowed, still I think we should downgrade EOPNOTSUPP unconditionally.

Maybe, we should document that if the socket or the kernel do not support a specified option, like PassCredentials=, we silently and gracefully ignore the failure.

---

Gracefully sure. I'm very wary of making this silent, because you then lose any trace of potential config issues. The proper solution is to create distinct sockets if options for specific types are employed, not to hide errors...

---

Or, maybe better to make the warning message more user friendly? Especially when multiple types of sockets configured.